ocrd_network/ocrd_network/runtime_data.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+from typing import Dict, List
+
+from .deployment_utils import (
+    create_docker_client,
+    create_ssh_client,
+    DeployType
+)
+
+__all__ = [
+    'DataHost',
+    'DataMongoDB',
+    'DataProcessingWorker',
+    'DataProcessorServer',
+    'DataRabbitMQ'
+]
+
+
+class DataHost:
+    def __init__(self, config: Dict) -> None:
+        self.address = config['address']
+        self.username = config['username']
+        self.password = config.get('password', None)
+        self.keypath = config.get('path_to_privkey', None)
+
+        # These flags are used to track whether a connection
+        # of the specified type will be required
+        self.needs_ssh: bool = False
+        self.needs_docker: bool = False
+
+        self.ssh_client = None
+        self.docker_client = None
+
+        # TODO: Not sure this is DS is ideal, seems off
+        self.data_workers: List[DataProcessingWorker] = []
+        self.data_servers: List[DataProcessorServer] = []
+
+        for worker in config['workers']:
+            name = worker['name']
+            count = worker['number_of_instance']
+            deploy_type = DeployType.DOCKER if worker['deploy_type'] == 'docker' else DeployType.NATIVE
+            if not self.needs_ssh and deploy_type == DeployType.NATIVE:
+                self.needs_ssh = True
+            if not self.needs_docker and deploy_type == DeployType.DOCKER:
+                self.needs_docker = True
+            for _ in range(count):
+                self.data_workers.append(DataProcessingWorker(self.address, deploy_type, name))
+
+        for server in config['servers']:
+            name = server['name']
+            port = server['port']
+            deploy_type = DeployType.DOCKER if server['deploy_type'] == 'docker' else DeployType.NATIVE
```

Comment:
This is needed to allow only workers or only servers (and small bugfix if deploy_type not provided as allowed in schema)
```suggestion
        for worker in config.get('workers', []):
            name = worker['name']
            count = worker['number_of_instance']
            deploy_type = DeployType.DOCKER if worker['deploy_type'] == 'docker' else DeployType.NATIVE
            if not self.needs_ssh and deploy_type == DeployType.NATIVE:
                self.needs_ssh = True
            if not self.needs_docker and deploy_type == DeployType.DOCKER:
                self.needs_docker = True
            for _ in range(count):
                self.data_workers.append(DataProcessingWorker(self.address, deploy_type, name))

        for server in config.get('servers', []):
            name = server['name']
            port = server['port']
            deploy_type = DeployType.DOCKER if server.get('deploy_type', None) == 'docker' else DeployType.NATIVE
```

Reply:
When I tested I provided one server without `deploy_type` specified. IIRC the spec allows that (because native is the default) and it failed with an exception. And my line 52 edits fixed that exception.